tools/benchmark/cmd/watch_latency.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2015 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	v3 "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/report"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+	"golang.org/x/time/rate"
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+// watchLatencyCmd represents the watch latency command
+var watchLatencyCmd = &cobra.Command{
+	Use:   "watch-latency",
+	Short: "Benchmark watch latency",
+	Long: `Benchmarks the latency for watches by measuring 
+	the latency between writing to a key and receiving the 
+	associated watch response.`,
+	Run: watchLatencyFunc,
+}
+
+var (
+	watchLTotal        int
+	watchLPutRate      int
+	watchLKeySize      int
+	watchLValueSize    int
+	watchLKeySpaceSize int
+	watchLSeqKey       bool
+)
+
+func init() {
+	RootCmd.AddCommand(watchLatencyCmd)
+	watchLatencyCmd.Flags().IntVar(&watchLTotal, "total", 10000, "Total number of watch responses.")
+	watchLatencyCmd.Flags().IntVar(&watchLPutRate, "put-rate", 100, "Number of keys to put per second")
+	watchLatencyCmd.Flags().IntVar(&watchLKeySize, "key-size", 32, "Key size of watch request")
+	watchLatencyCmd.Flags().IntVar(&watchLValueSize, "val-size", 32, "Val size of watch request")
+	watchLatencyCmd.Flags().IntVar(&watchLKeySpaceSize, "key-space-size", 1, "Maximum possible values")
+	watchLatencyCmd.Flags().BoolVar(&watchLSeqKey, "sequential-keys", false, "Use sequential values")
+}
+
+func watchLatencyFunc(cmd *cobra.Command, args []string) {
+	if watchLKeySpaceSize <= 0 {
+		fmt.Fprintf(os.Stderr, "expected positive --key-space-size, got (%v)", watchLKeySpaceSize)
+		os.Exit(1)
+	}
+
+	k := make([]byte, watchLValueSize)
+	if watchLSeqKey {
```

Comment:
This isn't what `sequential-keys` or `key-space-size` is supposed to do. It's meant to control what keys to use when the benchmark uses multiple keys. Since there's only one key, it doesn't make sense to have `--sequential-keys` or `--key-space-size`; probably just get rid of the options?

See https://github.com/coreos/etcd/blob/master/tools/benchmark/cmd/put.go#L93 for how it's used in practice.